test/lit/passes/precompute-gc.wast

```diff
@@ -1336,4 +1348,130 @@
    )
   )
  )
+
+ ;; CHECK:      (func $nested-struct-ref.eq-tee (type $2)
+ ;; CHECK-NEXT:  (local $A (ref $referrer))
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (block (result i32)
+ ;; CHECK-NEXT:    (drop
+ ;; CHECK-NEXT:     (local.tee $A
+ ;; CHECK-NEXT:      (struct.new $referrer
+ ;; CHECK-NEXT:       (struct.new_default $empty)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (i32.const 1)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $nested-struct-ref.eq-tee
+  (local $A (ref $referrer))
+  ;; As above, but immediately ref.eq the tee'd value with a get of itself. This
+  ;; can be computed to 1, but we must keep the tee effect.
+  (drop
+   (ref.eq
+    (local.tee $A
+     (struct.new $referrer
+      (struct.new_default $empty)
+     )
+    )
+    (local.get $A)
+   )
+  )
+ )
+
+ ;; CHECK:      (func $nested-struct-ref.eq-tee-2 (type $2)
+ ;; CHECK-NEXT:  (local $A (ref $referrer))
+ ;; CHECK-NEXT:  (local $B (ref $empty))
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (block (result i32)
+ ;; CHECK-NEXT:    (drop
+ ;; CHECK-NEXT:     (local.tee $A
+ ;; CHECK-NEXT:      (struct.new $referrer
+ ;; CHECK-NEXT:       (local.tee $B
+ ;; CHECK-NEXT:        (struct.new_default $empty)
+ ;; CHECK-NEXT:       )
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (i32.const 1)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $nested-struct-ref.eq-tee-2
```

Comment:
It would be interesting to test cases where the `ref.eq` evaluates to 0 as well.

Reply:
That already has coverage, see e.g. `$propagate-unequal` in this file.